specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+        Windows.Foundation.Collections.IPropertySet Values { get; };
+
+        /// Create or open the specified settings container in the settings container.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.createcontainer
+        ApplicationDataContainer CreateContainer(String name, ApplicationDataCreateDisposition disposition);
+
+        /// Delete the specified settings container, its subcontainers, and all settings in the hierarchy.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.deletecontainer
+        void DeleteContainer(String name);
+    };
+
+    /// Provides access to the application data store.
+    /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata
+    [contract(ApplicationDataContract, 1)]
+    runtimeclass ApplicationData : Windows.Foundation.IClosable
+    {
+        /// Get an instance of ApplicationData for the current user.
+        /// @note Requires the current process has package identity.
+        /// @note This is equivalent to Windows.Storage.ApplicationData.Current if running in an AppContainer.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily(GetCurrentPackageFamilyName()) if not running in an AppContainer.
+        static ApplicationData GetDefault();
+
+        /// Get an instance of ApplicationData for the specified user.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.GetForUserAsync()
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.getforuserasync
+        static ApplicationData GetForUser(Windows.System.User user);
+
+        /// Get an instance of ApplicationData for the specified package family for the current user.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily()
+        /// @see https://learn.microsoft.com/uwp/api/windows.management.core.applicationdatamanager.createforpackagefamily
+        static ApplicationData GetForPackageFamily(String packageFamilyName);
+
+        /// Get an instance of ApplicationData for the specified unpackaged app for the current user.
+        static ApplicationData GetForUnpackaged(String publisher, String name);
+
+        /// Return true if the package family supports the machine data store.
+        Boolean IsMachinePathSupported();
+
+        /// Return the path for the local cache data store not included in backup and restore operations.
+        /// @note This is equivalent to Windows.Storage.ApplicationDataManager.LocalCacheFolder().Path()
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.localcachefolder
+        String LocalCachePath { get; };
+
+        /// Return the path for the local data store. This location is backed up to the cloud.
```

Comment:
Are local paths really backed up to the cloud?

Reply:
Yes.

That's an exact quote from the docs for ApplicationData.LocalFolder.

LocalCacheFolder is similar to LocalFolder but it's _not_ backed up, as the docs mention:

>If the device that runs this app supports app data backup, any content placed on LocalFolder will be eligible to be backed up to the cloud. Therefore if you want to persist data and keep it local to the device you should consider using [LocalCacheFolder](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.localcachefolder?view=winrt-22621#windows-storage-applicationdata-localcachefolder).

If you have machine-specific data then you should use LocalCacheFolder, e.g. a license key for the Fabrikam MuffinMaker application which is tied to the machine it's installed on. No point having backup restore such data to another machine as such files could be restored but their content would be wrong for the other machine.

Reply:
Do you know of devices that support app data back up? I'm surprised by this.

Reply:
I dimly recall that we stopped backing them up, but I may have gotten that confused with roaming.

Reply:
RECOMMEND: Follow up with Windows.Storage.ApplicationData owner to possibly update the docs if the 'backup' comment is stale. If that changes also update here to match.